Update crawls list control bar UI

SuaYoo opened the pull request:
- Simplifies "Sort by" options (resolves https://github.com/webrecorder/browsertrix-cloud/issues/449)
- Adds crawl state (aka status) multi-select filter

### Manual testing
1. Run app and log in
2. Go to Crawls list view. Verify "View" filter defaults to "All Crawls" and "Sort by" defaults to "Date Created"
3. Select options from "View" filter. Verify crawls filter as expected
4. Sort crawls by dropdown options. Verify crawls are sorted as expected, and clicking the alternate sort icon with 2 arrows reverses the sort order.

### Screenshots
**Crawls list control bar:**
<img width="1118" alt="Screen Shot 2023-02-16 at 5 34 25 PM" src="https://user-images.githubusercontent.com/4672952/219512475-cc48d432-dfb8-4903-8c02-f6dd4a571577.png">

**Multiple status options selected:**
<img width="368" alt="Screen Shot 2023-02-16 at 5 35 47 PM" src="https://user-images.githubusercontent.com/4672952/219512521-f41cf77b-2860-4918-a530-3691f35edd25.png">

**Sort options:**
<img width="271" alt="Screen Shot 2023-02-16 at 5 36 35 PM" src="https://user-images.githubusercontent.com/4672952/219512594-3f2f2569-a8f3-4df9-bd98-27c3c354a49a.png">

**Empty state:**
<img width="1136" alt="Screen Shot 2023-02-16 at 5 35 58 PM" src="https://user-images.githubusercontent.com/4672952/219512624-d5fc3a79-e391-4adf-8a79-d853f8a005b0.png">

#### Differences from mockups:
- Added an empty state for when no crawls match -- could be improved & standardized later
- Using default Shoelace multi-select, which styles selected options into "tags" -- should be improved later
- No icons next to statuses, can be addressed with https://github.com/webrecorder/browsertrix-cloud/issues/608

SuaYoo committed:
update sort options

SuaYoo committed:
filter by crawl state

SuaYoo committed:
add empty state

SuaYoo committed:
update view options

SuaYoo committed:
update filter link

stavares843 reviewed (commented):
neat, minor comments 💥 

- should we display the filters when there isn't any data?

<img width="1263" alt="Captura de ecrã 2023-02-16, às 23 51 18" src="https://user-images.githubusercontent.com/29093946/219514047-8b68ca1c-4379-4a30-9c25-d2d5c6324baa.png">

- its in portuguese but it says in the top ''last month'' and in the bottom says ''4 weeks ago'', but 4 weeks ago is the same as last month right? should it show the same in both? either both ''last month'' or both ''4 weeks ago''

<img width="1229" alt="Captura de ecrã 2023-02-16, às 23 53 10" src="https://user-images.githubusercontent.com/29093946/219514159-93f7cdaf-bc0c-4bb2-b8c4-d2d22d442d83.png">

Shrinks99 commented:
Nice!

> should we display the filters when there isn't any data?

Our "No data" states across the app are in need of some rework.  In the future I'd like to do a 1 sprint pass in getting them all to the point where they offer the user actions to do which will be important as part of addressing our new user experience deficits.  Fine with leaving for this PR but it's on my radar!  This is likely a post-beta task.

---

The filter dropdown seems to take up a little more room than it needs to but is also slightly janky at some breakpoints.  Seems to be an overflow issue.  The search bar should be the element that shrinks to accommodate the others.

<img width="836" alt="Screenshot 2023-02-16 at 8 22 37 PM" src="https://user-images.githubusercontent.com/5672810/219525470-414632fe-11fe-470c-bcf0-b028e86a1865.png">

Shrinks99 committed:
Improves the crawl filter options CSS

- Grid is now set based on the amount of items in the bar
- Dropdowns have a max width set to around what the largest item / combination of items is possible.  This is possibly not ideal, @SuaYoo LMK
- Dropdowns are a single row on mobile instead of two full width rows
- Bottom switch is now aligned with flexbox instead of text-align for future compatibility with selection control buttons
-

Shrinks99 committed:
Puts the dropdowns in a flexbox

Wow terrible git dif because I added an indent, rad.

This replaces the grid settings with a flexbox for the dropdowns so they are aligned to the top of the box even when the dumb multi-select gets bigger due to simply not having enough space at the smallest breakpoint.  Is how it is, should be an edge case. :\

Shrinks99 committed:
Switches flexbox to CSS Grid

- Improves spacing at large breakpoints
- Resolves overflow issue and reverts to the 3 row view on mobile

Shrinks99 committed:
Removes redundant grid

A whole bunch of nonsense that is no longer required!

Shrinks99 commented:
I have fixed the CSS overflow issues in  https://github.com/webrecorder/browsertrix-cloud/tree/crawl-filter-controls-fix (created against your branch).  The grid is now set up with entries specifically for the items it contains.  The dropdowns have been set to a fixed width that encompasses all the possible largest options.  Sadly there's not enough room to put both dropdowns in the same row on mobile. :(

~~EDIT: Also it occurs to me that this max fixed width approach isn't ideal for localization reasons... I don't suppose there's a way to set it based on the longest item in the drop-down menu?  Likely not for the multi select anyways...~~

Fixed! :D

Large
<img width="1152" alt="Screenshot 2023-02-17 at 9 02 30 PM" src="https://user-images.githubusercontent.com/5672810/219826536-a3498940-8adc-4e84-bacb-a519b8f1c646.png">

Medium
<img width="890" alt="Screenshot 2023-02-18 at 1 18 52 PM" src="https://user-images.githubusercontent.com/5672810/219882025-c0396a29-716c-48bb-8098-c1657f67215a.png">

Small
<img width="443" alt="Screenshot 2023-02-17 at 11 59 12 PM" src="https://user-images.githubusercontent.com/5672810/219838900-5c375cd3-ace5-4f82-8406-87a996470f9c.png">

Other changes include:
- Switched from text-align to flexbox for the filter switch — Will be used in the future for multiple crawl selection, deletion, & tagging (planned but hidden currently in the mockups!)
- Added a 0.5rem top value for position sticky so it's not right up against the top of the browser when scrolling

PR for these changes into this branch: https://github.com/webrecorder/browsertrix-cloud/pull/615

Shrinks99 committed:
Improves responsiveness at medium breakpoint

- Two row layout to give the search field more space, should be better future proofing for eventual filter editor & also now the search placeholder text doesn't get hidden.

Shrinks99 committed:
Fixes broken tab / space shenanigans

Shrinks99 committed:
remove row-start-1

Shrinks99 committed:
Update frontend/src/pages/org/crawls-list.ts

Co-authored-by: sua yoo <sua@webrecorder.org>

Shrinks99 committed:
removes incorrectly set margin

Shrinks99 committed:
Switches to min-width to avoid text getting cut off

Shrinks99 committed:
Update frontend/src/pages/org/crawls-list.ts

Co-authored-by: sua yoo <sua@webrecorder.org>

Shrinks99 committed:
Update frontend/src/pages/org/crawls-list.ts

Co-authored-by: sua yoo <sua@webrecorder.org>
